Sources/Citadel/OpenSSHKey.swift

```diff
@@ -0,0 +1,223 @@
+import CCryptoBoringSSL
+import BigInt
+import Foundation
+import NIO
+import Crypto
+import NIOSSH
+
+internal protocol ReadableFromBuffer {
+    static func read(from buffer: inout ByteBuffer) throws -> Self
```

Comment:
Maybe while we're at it, change the label `from` to `consuming`. It's really easy to make API's using common words as label like `from`, `in`, and `to`. But those labels can be powerful tools in clarifying an API's intent.